libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -687,6 +687,105 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_curvilinear_get_boundaries_as_polygons(int meshKernelId, int lowerLeftN, int lowerLeftM, int upperRightN, int upperRightM, GeometryList& boundaryPolygons)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+
+            if (!meshKernelState[meshKernelId].m_curvilinearGrid->IsValid())
+            {
+                throw meshkernel::MeshKernelError("Invalid curvilinear grid");
+            }
+
+            if (lowerLeftN < 0)
```

Comment:
Those assertions will probably never fire. If you add these assertions then also add assertions that indexes cannot be larger than MaxM & MaxN, and that lower-left indexes must be less than corresponding upper-right indexes. Alternatively, min/max computations can be done here (and even indexes can be clamped).